soroban-sdk/src/env.rs

```diff
@@ -683,14 +689,43 @@ impl Env {
             auth: Default::default(),
         };
 
-        let token_id = self.env_impl.invoke_functions(vec![create]).unwrap()[0]
+        let token_id: BytesN<32> = self.env_impl.invoke_functions(vec![create]).unwrap()[0]
             .try_into_val(self)
             .unwrap();
+
+        let source_account = self.env_impl.source_account();
+        self.env_impl.set_source_account(issuer_id);
+
+        // Recording auth is used with set_admin because account auth is
```

Comment:
The comment doesn't correspond to what happens here - we seem to be using enforcing mode with invoker auth.

Reply:
Thanks for catching that. I used recording auth then switched to using the actually issuer since the auth was simple and convenient to construct.

Reply:
Just a reminder, I still don't see the update.

Reply:
Thanks. Fixed in c17bc60.